### ISSUE TITLE
Incorrect(?) download link for "unlabelled candidates"/"full dataset" + Label space clarification

### ISSUE THREAD

**Issue by jwkirchenbauer** [peterwestai2/symbolic-knowledge-distillation#5]
Awesome paper! - Excited to try out the pipeline myself

1) In walking through your documentation I was taking a look at the two links on the [purification_code](https://github.com/peterwestai2/symbolic-knowledge-distillation/tree/main/purification_code), and I downloaded both. As expected, `purification_dataset.jsonl.zip` seems to be the 10K row training dataset selected for training the critic model, but `unique_data.jsonl.zip` seems to contain the ~6.5M rows of what look like filtered results? which matches the name, but I think the data that should be here in the documentation is "The unlabelled candidates" before filtering? I saw that the example command in the `predict.py` file itself seems to suggest the file/name should be `full_dataset_unique_examples.jsonl`. I also note that the `unique_data.jsonl` file looks identical to `ATOMIC10X.jsonl`. So I'm checking as to whether we're missing a link to a dataset here? Would the dataset be the same size before and after? It looks as if the script is providing model scores in field `p_valid_model` but these files already have that field.

If I'm completely missing something here and the above makes no sense, then I would just appreciate help parsing the meaning of the `rec_#.#` fields and origin of the `p_valid_model` in an example row from `unique_data.jsonl`/`ATOMIC10X.jsonl`

```
{
 "head": "PersonX decides to see a therapist", 
 "relation": "xEffect", 
 "tail": "feels better", 
 "split": "train", 
 "rec_0.6": true, "rec_0.9": true, "rec_0.5": true, "rec_0.7": true, "rec_0.8": true, 
 "p_valid_model": 0.9934503436088562
}
```

2) What is the human interpretation of the the label space for the purification dataset -`purification_dataset.jsonl`? I saw {-9, -2,-1, 1, 2}, didn't check for completeness, and I do note that you binarize in the [`classify.py#L70`](https://github.com/peterwestai2/symbolic-knowledge-distillation/blob/main/purification_code/classify.py#L70), I was just curious

**Comment by peterwestai2** [comment]
Hi! Thank you for the interest and questions

To your first question, the size before and after will actually be the same. This exactly has to do with the `rec_#.#` fields that you bring up. The purification code (corresponding to the critic from the paper) gives a score to each example, of how likely it is "acceptable" (based on the training data you mentioned). This score is `p_valid_model`, where a higher score indicates the model predicts a higher acceptability.

using `p_valid_model`, we can generate filtered corpora by picking a threshold and rejecting everything below this threshold. So the each `rec_#.#` indicates whether the given example is rejected at different thresholds that we have preselected. `rec_#.#` specifically refers to a threshold that gives an estimated recall of `#.#`. So for instance, if you only take all examples for which `rec_0.9` is true (i.e. filter all others), you should end up with a filtered dataset that retains 0.9 of all acceptable examples, but has a higher precision than the full corpus. So in summary, the filtered dataset file will have the same number of examples as the full dataset, but will indicate which examples to remove to get various filtered versions.

I will ensure that the files you mentioned above are correct though, and get back to you soon!

To your second question, The scores correspond to:
2: This assertion always/often holds
1: This assertion sometimes holds/is likely
-1: The assertion is farfetched/never holds
-2: The assertion is invalid
-9: The assertion is too unfamiliar for the human annotator to judge

We include the human annotation template in the appendix of the paper if you would like to get a better sense of these!

**Comment by jwkirchenbauer** [comment]
Thanks for the detailed reply!

So to clarify, the `predict.py` script of the purification code, if fed in a file (say a more "raw" copy of `ATOMIC10X.jsonl`) without `rec_#.#`'s or `p_valid_model`, will take the critic model, make predictions to fill in the `p_valid_model` field, and in a second pass or something, add the `rec_#.#` fields to help filter the dataset size versus estimated quality (as in the paper) w.r.t. to the scores produced by the specific critic model presented.

I think I was just confused by the filename in this command 
`python predict.py model~model=roberta-large-mnli... results~model=roberta-large-mnli... unique_data.jsonl`
because the data files including that one (plus `ATOMIC10x.jsonl`),  which was what I assumed would be the "raw" large dataset, already have those fields. But I'm guessing you just left them in for convenience/auditing

**Comment by peterwestuw** [comment]
You are correct about the intended functionality of predict.py! In general, the intention is that a user can define their own cutoffs for `p_valid_model`, with the general trend of a higher cutoff giving lower recall and higher precision.

And yes, that command is definitely confusing w.r.t. `ATOMIC10X.jsonl`. If I remember correctly (will verify) I believe that command results in individual dataset files for cutoffs calculated within the script. So one way to get a file like `ATOMIC10X.jsonl` after the fact is to fold these together into 1 file, with `rec_#.#` indicating whether a given example appears in a given filtered dataset. But lines [lines 130 to 134](https://github.com/peterwestai2/symbolic-knowledge-distillation/blob/973f0a00024c1b9aec1e5a9f571bdea059ede323/purification_code/predict.py#L130-L134) should output a separate file with the full set of examples, including `p_valid_model` for each one. 

You are also correct that we include those fields in `ATOMIC10X.jsonl` in general for convenience, to have one large dataset file that contains enough information to easily derive the datasets from the paper, or even define new ones based on new thresholds

**Comment by peterwestai2** [comment]
You are correct about the intended functionality of predict.py! In general, the intention is that a user can define their own cutoffs for `p_valid_model`, with the general trend of a higher cutoff giving lower recall and higher precision.

And yes, that command is definitely confusing w.r.t. `ATOMIC10X.jsonl`. If I remember correctly (will verify) I believe that command results in individual dataset files for cutoffs calculated within the script. So one way to get a file like `ATOMIC10X.jsonl` after the fact is to fold these together into 1 file, with `rec_#.#` indicating whether a given example appears in a given filtered dataset. But lines [lines 130 to 134](https://github.com/peterwestai2/symbolic-knowledge-distillation/blob/973f0a00024c1b9aec1e5a9f571bdea059ede323/purification_code/predict.py#L130-L134) should output a separate file with the full set of examples, including `p_valid_model` for each one. 

You are also correct that we include those fields in `ATOMIC10X.jsonl` in general for convenience, to have one large dataset file that contains enough information to easily derive the datasets from the paper, or even define new ones based on new thresholds

**Comment by peterwestai2** [comment]
Closing this for now. Feel free to reopen if we missed anything
